### DISCUSSION TITLE
upgrade error

### DISCUSSION THREAD

**Discussion by castelhanod** [gravitational/teleport#6408] [Q&A] [answered]
Hi,
After upgrading from 5.1.0-1 to 6.1.0-1 on rhel7, the teleport service enters a systemd crash/restart loop with the journalctl error snippet pasted below. The config has been unchanged from previous 4.x and 5.x upgrades, and was working correctly before upgrading to 6.1. This node runs all teleport services (auth/proxy/ssh), and is the primary trusted cluster. For the other clusters, one is running 6.1 (ubuntu), and the other 5.1.2 (rhel7).

Any insight on what to check to resolve?

Thank you,
Dan

Apr 12 10:18:35 ServerName1 teleport[6441]: INFO [PROC:1]    The new service has started successfully. Starting syncing rotation status with period 10m0s. service/connect.go
:433
Apr 12 10:18:35 ServerName1 teleport[6441]: panic: merger not found for type:int
Apr 12 10:18:35 ServerName1 teleport[6441]: goroutine 978 [running]:
Apr 12 10:18:35 ServerName1 teleport[6441]: github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo(0xc0000bd500)
Apr 12 10:18:35 ServerName1 teleport[6441]: /go/src/github.com/gravitational/teleport/vendor/github.com/gogo/protobuf/proto/table_merge.go:662 +0xfe5
Apr 12 10:18:35 ServerName1 systemd[1]: teleport.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 12 10:18:35 ServerName1 systemd[1]: Unit teleport.service entered failed state.
Apr 12 10:18:35 ServerName1 teleport[6441]: /go/src/github.com/gravitational/teleport/lib/reversetunnel/remotesite.go:342 +0xbf8
Apr 12 10:18:35 ServerName1 teleport[6441]: created by github.com/gravitational/teleport/lib/reversetunnel.(*server).handleNewCluster
Apr 12 10:18:35 ServerName1 teleport[6441]: /go/src/github.com/gravitational/teleport/lib/reversetunnel/srv.go:691 +0x265
Apr 12 10:18:35 ServerName1 systemd[1]: teleport.service failed.
Apr 12 10:18:36 ServerName1 systemd[1]: teleport.service holdoff time over, scheduling restart.                                                                              -- Subject: Unit teleport.service has finished shutting down
-- Unit teleport.service has finished shutting down.

**Comment by castelhanod** [comment]
I've narrowed it down to the trusted cluster configuration. If I remove the tc/rc definition on the root and leaf clusters, the main cluster will start on 6.13, and I can access the nodes connected to the main cluster.

The main cluster crashes immediately on service startup the second I add the leaf cluster, and systemd attempts to start teleport every time it crashes.

Any ideas as to what I may have wrong on the main cluster config to cause an error adding a trusted cluster?

Thanks!

**Comment by webvictim** [comment, marked as answer]
This is a bug that was fixed in `master` in https://github.com/gravitational/teleport/pull/5712

A [backport is currently open to the v6.1 branch](https://github.com/gravitational/teleport/pull/6687) and so this should be fixed in the next 6.1.x Teleport release.

**Reply by castelhanod** [reply]
resolved with version 6.1.5. thanks!

**Reply by webvictim** [reply]
Great news!
